smithy-typescript-codegen/src/test/java/software/amazon/smithy/typescript/codegen/TypeScriptSettingsTest.java

```diff
@@ -87,6 +98,121 @@ private static Stream<Arguments> providePackageDescriptionTestCases() {
         );
     }
 
+    @Test
+    public void resolveServiceProtocol(@Mock Model model,
```

Comment:
Consider instead using parametrized tests. This test is testing all the cases instead of having a single test method for each.

Reply:
Split the spec cases out into separate test methods